### ISSUE TITLE
 打开enable_cluster_plan，update时条件中临时表和非临时表join，报错 fail to fetch slot from self reduce

### ISSUE THREAD

**Issue by mass3** [ADBSQL/AntDB#41]
### 打开enable_cluster_plan，update时条件中临时表和非临时表join，报错 fail to fetch slot from self reduce
```
postgres=# set enable_cluster_plan = on;
SET
postgres=# CREATE TABLE writetest (a int);
CREATE TEMPORARY TABLE temptest (a int);
UPDATE temptest SET a = 0 FROM writetest WHERE temptest.a = 1 AND writetest.a = temptest.a;
CREATE TABLE
postgres=# CREATE TEMPORARY TABLE temptest (a int);
CREATE TABLE
postgres=# UPDATE temptest SET a = 0 FROM writetest WHERE temptest.a = 1 AND writetest.a = temptest.a;
ERROR:  fail to fetch slot from self reduce
DETAIL:  the peer of [REDUCE 11990] has performed an orderly shutdown
postgres=# 
```

```
CREATE TABLE writetest (a int);
CREATE TEMPORARY TABLE temptest (a int);
UPDATE temptest SET a = 0 FROM writetest WHERE temptest.a = 1 AND writetest.a = temptest.a;

```

**Comment by hsiaoby** [comment]
like #31, fixed by 850072921edc2f48129cf8217f106bbc761c28a3

**Comment by mass3** [comment]
update结果正确
```
postgres=# CREATE TABLE writetest (a int);
CREATE TEMPORARY TABLE temptest (a int);
UPDATE temptest SET a = 0 FROM writetest WHERE temptest.a = 1 AND writetest.a = temptest.a;
CREATE TABLE
postgres=# CREATE TEMPORARY TABLE temptest (a int);
CREATE TABLE
postgres=# UPDATE temptest SET a = 0 FROM writetest WHERE temptest.a = 1 AND writetest.a = temptest.a;
UPDATE 0
postgres=# 

```
